### PR TITLE
Remove gdal and other optional dependency

### PR DESCRIPTION
**Pull request by xylar**
It should be optional.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
I saw mentioned here: https://github.com/SciTools/iris/pull/3762#issuecomment-685573007
that `gdal` should be an optional dependency, and that's clearly true:
https://github.com/SciTools/cartopy/blob/v0.18.0/requirements/plotting.txt

I'm not sure why I added it as required in #88.  Maybe I was just confused about which were required and which not.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I added not only `gdal` but several other dependencies in #88 that should have been optional.  Sorry! 😭😭😭

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge/cartopy, sorry for the mess in #88.  I think this is ready for review if anyone cares to take a look.  Based on that experience, another pair of eyes would be a good idea.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ source:
 
 build:
   skip: true  # [win and vc<14]
```
Comment (isuruf) [inline recipe/meta.yaml:12]: To avoid these issues in the future
```suggestion
  skip: true  # [win and vc<14]
  error_overlinking: true
```

Comment (xylar) [reply recipe/meta.yaml:12]: @isuruf, thanks, I didn't know about that option.

**Comment by bjlittle** [comment]
@xylar Thanks for doing this. The gdal package is a monster and ultimately would have caused dependency issues further down the line for cartopy in a conda environment - particularly as gdal pins the version of libnetcdf, which can restrict other packages from installing their latest versions 👍

**Comment by xylar** [comment]
I'm going ahead with the merge.  Presumably, this won't make things worse than they currently are...

**Comment by danschef** [comment]
@xylar @bjlittle 
This PR removes pyepsg from the run requirements although it is still imported in _epsg.py (see [here](https://github.com/SciTools/cartopy/blob/d12c86ca228365c61d5fb8b7121b49201f19fea8/lib/cartopy/_epsg.py#L29)).

This leads to breaking code, e.g., when using the ccrs_from_epsg() function.

**Comment by xylar** [comment]
@danschef, I'm pretty sure you need to explicitly include `pyepsg` as a dependency if you want to use `ccrs_from_epsg()`.  It is an optional dependency for those not using that code.  Similarly, a lot of plotting won't work without `matplotlib` but it is still technically optional.

**Comment by xylar** [comment]
@danschef, however, feel free to open an issue and we can discuss this further.

**Comment by danschef** [comment]
I see, it is listed in the optional dependencies in the documentation. However, it would be nice if cartopy would raise a helpful exception in case optional deps are not installed - instead of just raising an ImportError. I will open an issue for that.

**Comment by xylar** [comment]
@danschef, that would be an issue for the cartopy repo, this is just a recipe for building cartopy for the conda-forge channel.

**Comment by danschef** [comment]
Yes, I know.

**Comment by mathause** [comment]
I find it a bit unfortunate that you did this without a version bump. Because stuff that used to work in version 0.18 now doesn't work in 0.18...

Yes, I know I can easily add the dependencies. But my package doesn't, so I may have to issue a new release there (of course a new version wouldn't have helped my package). Anyway, sorry for the rant & thanks for all your work on cartopy.

**Comment by xylar** [comment]
@mathause, you need to install some of the optional dependencies for your work.  We don't determine the version number at conda-forge, it's determined upstream.

**Comment by xylar** [comment]
@mathause, sorry for the terse answer earlier, and sorry that this change caused you trouble.  The original build of 0.18.0 incorrectly included a lot of optional packages, and several folks were complaining about that.  I have tried to contain the damage but clearly it's causing trouble for you the other direction.

Would it be possible in the context you're working in to just add the optional dependencies along with the `cartopy` package?

**Comment by mathause** [comment]
Thanks, appreciate it. I also didn't mean to be rude, apologies!

I did not know matplotlib & scipy were optional dependencies of cartopy - they were always included via conda (e.g. `conda search --info cartopy=0.13`). So I was surprised. I'll be able to work around it & think that this can be beneficial in the long term.

**Comment by ocefpaf** [comment]
@xylar and @bjlittle looking into this I'm not sure removing matplotlib makes sense. After all cartopy's third point on its overview is:

    integration to expose advanced mapping in Matplotlib with a simple and intuitive interface

**Comment by xylar** [comment]
@ocefpaf, and yet, they have gone out of their way to make `matplotlib` an optional dependency...

**Comment by dopplershift** [comment]
While it *is* an optional dependency, I'm also not at all clear what utility CartoPy has in that case--or whether anyone is actually using it in that way.

**Comment by xylar** [comment]
I don't have a problem with adding `matplotlib-base` back as a dependency.  I think it was `gdal` that folks were wanting to be optional.  But I figured once I took that out I should follow the lead of the developers upstream...

**Comment by ocefpaf** [comment]
@dopplershift what do you think is appropriate for the conda package?

@xylar indeed, `gdal` was too much! That one must remain an optional dep.

PS:I'd love to heat what @bjlittle thinks about this too.

**Comment by dopplershift** [comment]
I think a dependency on `matplotlib-base` is appropriate and what most users would expect. We never heard any complaints in the first 4 years. ;)

**Comment by xylar** [comment]
@ocefpaf and @dopplershift, I don't mind making a PR to add back `matplotlib-base`.  From upstream, the plotting requirements are:
```
matplotlib>=1.5.1
GDAL>=1.10.0
pillow>=1.7.8
scipy>=0.10
```
We've already said that `gdal` is out because it's too big.  What about `pillow` and `scipy`?  There's not indication I can see from upstream of how these requirements can be mixed and matched.  I would have assumed you need all 4 or plotting wouldn't work.  But I certainly haven't experimented with that.

**Comment by ocefpaf** [comment]
I would add only `matplotlib-base`. scipy is for the KDTree stuff, pillow and gdal to handle images and raster transformation stuff. People doing those will actively install them regardless of cartopy. However, people wanting to make a map may try to install only cartopy just to find out they also need matplotlib. (Optional deps are hard!)

**Comment by xylar** [comment]
@ocefpaf, thanks for following up on this and for understanding the dilemma I was facing.

**Comment by ocefpaf** [comment]
> @ocefpaf, thanks for following up on this and for understanding the dilemma I was facing.

Don't worry. I'm still on not 100% sure what is the best course for us here. I'm 80% on the site of leaving mpl b/c that is what all the users I know expect. Still, that is a subset of all the users.

**Comment by xylar** [comment]
I 100% agree that it's hard for me to imagine why anyone would install `cartopy` without `matplotlib` and I don't know of anyone who would want that.  So I don't see any downside to adding the requirement back in, especially because no one as complained about it before and folks *did* complain when it was gone.

I'm curious why it's not a default requirement upstream but not curious enough to follow up there.

**Pull request merged**